demos/plasmas/ionisation_balance_2d.py

```diff
@@ -106,13 +106,13 @@ def exp_decay(r, lamb, max_val, lim=None):
 n_element2_profile_1d = double_parabola(psin_1d, 5e17, 1e17, 2, 2, 1)
 n_tcx_donor_profile_1d = exp_decay(psin_1d, 10, 3e16, 1)
 
-t_e_1d = Interpolate1DCubic(psin_1d, t_e_profile_1d)
-n_e_1d = Interpolate1DCubic(psin_1d, n_e_profile_1d)
+t_e_1d = Interpolator1DArray(psin_1d, t_e_profile_1d, 'linear', 'none', 0)
+n_e_1d = Interpolator1DArray(psin_1d, n_e_profile_1d, 'linear', 'none', 0)
 
-t_element_1d = Interpolate1DCubic(psin_1d, t_element_profile_1d)
-n_element_1d = Interpolate1DCubic(psin_1d, n_element_profile_1d)
-n_element2_1d = Interpolate1DCubic(psin_1d, n_element2_profile_1d)
-n_tcx_donor_1d = Interpolate1DCubic(psin_1d, n_tcx_donor_profile_1d)
+t_element_1d = Interpolator1DArray(psin_1d, t_element_profile_1d, 'linear', 'none', 0)
+n_element_1d = Interpolator1DArray(psin_1d, n_element_profile_1d, 'linear', 'none', 0)
+n_element2_1d = Interpolator1DArray(psin_1d, n_element2_profile_1d, 'linear', 'none', 0)
+n_tcx_donor_1d = Interpolator1DArray(psin_1d, n_tcx_donor_profile_1d, 'linear', 'none', 0)
```

Comment:
Why not leave these as cubic?

Reply:
It appears that the bug was in `lsq_linear`, https://github.com/scipy/scipy/issues/9982. This has now been fixed, so everything works fine with the recent scipy release. I rolled back to cubic splines in this demo.

Reply:
I was going to say, we are not responsible for lsq_linear!